### DISCUSSION TITLE
Conditional refresh?

### DISCUSSION THREAD

**Discussion by kleynjan** [zauberzeug/nicegui#3415] [Q&A] [answered]
Just a quick check before I get in too deep with my current project: if a @refreshable is refreshed, **all** of its instances are recreated. There is currently no mechanism in NiceGUI to limit the scope of this refresh cycle -- correct? 

Has anyone already done work on this, eg, executing the refresh based on state logic?

**Comment by rodja** [comment, marked as answer]
There are at least three variants to get local scope: Please start reading here: https://nicegui.io/documentation/refreshable#local_scope_(variant_a)

**Reply by kleynjan** [reply]
Oops @rodja, somehow early on my mental model of NiceGUI acquired a global registry of refreshables... I had seen the docs you refer to but falsely assumed they only scoped the visibility of the function.

Thanks!
